src/redux/slices/user.js

```diff
@@ -213,27 +216,107 @@ export const transferNFT = createAsyncThunk(
   }
 );
 
+export const getContacts = createAsyncThunk(
+  'keyring/getContacts',
+  async (walletNumber = 0, { getState, dispatch }) => {
+    try {
+      dispatch(setContactsLoading(true));
+      const state = getState();
+      const res = await state.keyring.instance?.getContacts(walletNumber);
+      return res?.map(formatContact);
+    } catch (e) {
+      dispatch(setContactsLoading(false));
+      console.log('Error getting contacts:', e);
+    }
+  }
+);
+
+export const addContact = createAsyncThunk(
+  'keyring/addContact',
+  async ({ contact, walletNumber = 0, onFinish }, { getState, dispatch }) => {
```

Comment:
is it OK to set walletNumber = 0 as default?

Reply:
Yes! Its totally needed!